pkg/common/aws/s3.go

```diff
@@ -85,3 +88,50 @@ func ParseS3URL(s3URL string) (string, string, error) {
 
 	return parsedURL.Host, parsedURL.Path, nil
 }
+
+// CleanupS3Buckets finds buckets with prefix "osde2e-" or "managed-velro"
+// older than given duration, then deletes bucket objects and then buckets
+func (CcsAwsSession *ccsAwsSession) CleanupS3Buckets(olderthan time.Duration, dryrun bool) error {
+	err := CcsAwsSession.GetAWSSessions()
+	if err != nil {
+		return err
+	}
+
+	input := &s3.ListBucketsInput{}
+	result, err := CcsAwsSession.s3.ListBuckets(input)
+	if err != nil {
+		return err
+	}
+
+	for _, bucket := range result.Buckets {
+		if (strings.Contains(*bucket.Name, rolesubstr) || strings.Contains(*bucket.Name, velerosubstr)) && time.Since(*bucket.CreationDate) > olderthan {
+			fmt.Printf("Bucket will be deleted: %s\n", bucket)
+			if !dryrun {
+				// Setup BatchDeleteIterator to iterate through a list of objects.
+				iter := s3manager.NewDeleteListIterator(CcsAwsSession.s3, &s3.ListObjectsInput{
+					Bucket: bucket.Name,
+				})
+				if err := s3manager.NewBatchDeleteWithClient(CcsAwsSession.s3).Delete(aws.BackgroundContext(), iter); err != nil {
+					return err
+				} else {
+					fmt.Println("Deleted object(s) from bucket")
+
+					input := &s3.DeleteBucketInput{
+						Bucket: bucket.Name,
+					}
+					// Delete the policy
+					_, err := CcsAwsSession.s3.DeleteBucket(input)
+					if err != nil {
+						return err
+					}
+					fmt.Println("Deleted bucket")
+
+				}
+			}
+		} else {
+			fmt.Printf("Bucket will NOT be deleted: %s\n", *bucket.Name)
+		}
+	}
+
+	return nil
```

Comment:
I think we can simplify the conditions just a bit like this:

```suggestion
    if err := CcsAwsSession.GetAWSSessions(); err != nil {
        return err
    }

    result, err := CcsAwsSession.s3.ListBuckets(&s3.ListBucketsInput{})
    if err != nil {
        return err
    }

    batchDeleteClient := s3manager.NewBatchDeleteWithClient(CcsAwsSession.s3)

    for _, bucket := range result.Buckets {
        if !((strings.Contains(*bucket.Name, rolesubstr) || strings.Contains(*bucket.Name, velerosubstr)) && time.Since(*bucket.CreationDate) > olderthan) {
            fmt.Printf("Skipping bucket: %s\n", bucket)
            continue
        }

        if !dryrun {
            iter := s3manager.NewDeleteListIterator(CcsAwsSession.s3, &s3.ListObjectsInput{Bucket: bucket.Name})
            if err := batchDeleteClient.Delete(aws.BackgroundContext(), iter); err != nil {
                return err
            }
            fmt.Printf("Deleted object(s) from bucket: %s\n", bucket)
            if _, err := CcsAwsSession.s3.DeleteBucket(&s3.DeleteBucketInput{Bucket: bucket.Name}); err != nil {
                return err
            }
        }
        fmt.Printf("Deleted bucket: %s\n", bucket)
    }

    return nil

```

Reply:
Please clean up the logs as well

Reply:
In my suggestion, I've updated the log messages to include the bucket name to make it clearer